controllers/openstackvmset_controller.go

```diff
@@ -281,22 +273,39 @@ func (r *OpenStackVMSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
-	for _, net := range instance.Spec.Networks {
+	for _, netNameLower := range instance.Spec.Networks {
 		timeout := 10
 		var msg string
 		actualProvisioningState.State = ospdirectorv1beta1.VMSetWaiting
 
-		if osNetBindings[net] == "" {
+		// get network with name_lower label
+		network, err := openstacknet.GetOpenStackNetWithLabel(
+			r,
+			instance.Namespace,
+			map[string]string{
+				openstacknet.SubNetNameLabelSelector: netNameLower,
+			},
+		)
```

Comment:
I know you use this new function a lot.  Is it essentially a convenience function so you don't have to somehow convert the `NameLower` into the `Name` of the target resource?  I'm guessing that might be impossible anyhow.

Reply:
yes, because the osnet CR name must not be the NameLower name, e.g NameLower can have '_', while this is a forbidden char in CR names. That's why I add the ooo Name, NameLower, SubnetName as labels to the osnet CRs and then can use the ooo names to get the correct osnet.